0003-lightning-as-alpha-ledger.adoc

```diff
@@ -0,0 +1,354 @@
+=   LND as Alpha Ledger as described in RFC003
+
+NOTE: Authors: Philipp Hoenisch +
+Date: 2019-04-24 +
+Tracking issue: TBA
+
+:toc:
+
+== Context
+
+The **Hold invoice feature** was introduced in the following PR: https://github.com/lightningnetwork/lnd/pull/2022[lightningnetwork/lnd/pull/2022].
+____
+A hold invoice is a new type of invoice that triggers a different flow at the receiver's end.
+Instead of immediately locking in and settling the htlc when the payment arrives, the htlc for a hold invoice is only locked in and not yet settled.
+At that point, it is not possible anymore for the sender to revoke the payment, but the receiver still can choose whether to settle or cancel the htlc and invoice.
+____
+
+This allows us to trade Bitcoin on the Lightning Network (LN) for an asset _X_ on a different ledger where LN is the _alpha ledger_ as per the definition in https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.md[RFC003].
+For the other way around, Bitcoin on the Lightning Network is the _beta asset/ledger_, we can make use of `createinvoice` as Alice would instantly release the secret by settling Bob's LN payment, or we can follow the same approach as described below.
+
+NOTE: Within this spike we only focus on LN as the Alpha Ledger using https://github.com/lightningnetwork/lnd/[LND].
+
+== Research
+
+=== Standard _hold invoice_ process
+
+____
+Assumption: Alice wants to pay Bob _X_ Bitcoin using the LN but Bob does not generate the preimage - Alice does it.
+____
+
+1) Alice generates a secret preimage `p` and sends the hash of it `h(p)` to Bob.
+2) Bob calls `addholdinvoice` with an amount `X` and the hashed preimage `h(p)` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L142[more details]).
+3) Bob sends the generated invoice to Alice.
+4) Alice now can pay the invoice using `sendpayment`.
+5) As soon as Alice is happy (because she is in a good mood), she tells Bob the secret (or he learns it otherwise, see below).
+6) Bob settles the payment by calling `settleInvoice` (https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L53[more details]).
+
+The payment is complete.
+
+== Lightning Network as Alpha Ledger within the bounds of RFC003
+
+____
+Assumption: Bob wants to receive _X_ Bitcoin (**A** Asset) on the LN (**alpha** Ledger) for _Y_ Ether (**B** Asset) on Ethereum (**beta** Ledger)
+Alice magically finds out about this _wish_ or using a COMIT link.
+
+* Alpha Ledger: Lightning Network
+* Alpha Asset: Bitcoin
+* Beta Ledger: Ethereum
+* Beta Asset: Ether
+____
+
+According to https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.md[RFC003], Alice needs to send the following information to Bob for initiating a swap.
```

Comment:
The confusion should be cleared at the end of this section:
> Because of these limitations, we should introduce a new protocol for using LN as Alpha Ledger.

do you want me to mention this already at the beginning of the section?

Reply:
ok, will do so as discussed offline.